mesh/nurbs.cpp

```diff
@@ -2770,6 +2770,120 @@ void NURBSExtension::Generate3DElementDofTable()
    el_dof = new Table(NumOfActiveElems, el_dof_list);
 }
 
+int NURBSExtension::NumPatchDofs1D(const int patch, const int dim)
+{
+   if (ndof1D.NumRows() <= patch)
+   {
+      GeneratePatchDofTable();
```

Comment:
This is a possibility, but I avoided it for the following reasons. 

1. There are many constructors that need to call the same function. This can be done if it is clearly advantageous for performance.
2. Only the 3D case is supported so far. We could do the setup only for 3D in the constructors, but `GetPatchDofs` will still have to verify the 3D case. This is not faster than checking whether the table has been set already.
3. The table `patchDofs` has a significant amount of data, and in most use cases of NURBS it is not necessary. So I am taking this lazy construction approach to avoid constructing `patchDofs` when it is not requested.

Reply:
Clear to me!